glyphs-reader/src/font.rs

```diff
@@ -125,129 +230,340 @@ struct RawFont {
     pub manufacturerURL: Option<String>,
     pub versionMajor: Option<i64>,
     pub versionMinor: Option<i64>,
-    pub axes: Option<Vec<Axis>>,
+    pub axes: Vec<Axis>,
     pub glyphs: Vec<RawGlyph>,
     pub font_master: Vec<RawFontMaster>,
-    #[fromplist(default)]
     pub instances: Vec<RawInstance>,
-    pub feature_prefixes: Option<Vec<RawFeature>>,
-    pub features: Option<Vec<RawFeature>>,
-    pub classes: Option<Vec<RawFeature>>,
-    pub properties: Option<Vec<RawName>>,
+    pub feature_prefixes: Vec<RawFeature>,
+    pub features: Vec<RawFeature>,
+    pub classes: Vec<RawFeature>,
+    pub properties: Vec<RawName>,
+    #[fromplist(alt_name = "kerning")]
+    pub kerning_LTR: Kerning,
+    pub custom_parameters: CustomParameters,
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct CustomParameters(BTreeMap<String, CustomParameterValue>);
+
+impl CustomParameters {
+    fn int(&self, name: &str) -> Option<i64> {
+        let Some(CustomParameterValue::Int(i)) = self.0.get(name) else {
+            return None;
+        };
+        Some(*i)
+    }
 
-    #[fromplist(rest)]
-    pub other_stuff: BTreeMap<String, Plist>,
+    fn bool(&self, name: &str) -> Option<bool> {
+        self.int(name).map(|v| v == 1)
+    }
+
+    fn string(&self, name: &str) -> Option<&str> {
+        let Some(CustomParameterValue::String(str)) = self.0.get(name) else {
+            return None;
+        };
+        Some(str)
+    }
+
+    fn axes(&self) -> Option<&Vec<Axis>> {
+        let Some(CustomParameterValue::Axes(axes)) = self.0.get("Axes") else {
```

Comment:
not being familiar with the format, I have to do a lot of reading between the lines to figure out what's probably going on here; I presume in some cases there can be arbitrary keys/values, but in other cases there are defined keys that have expected types? Is there any reason that `CustomParameters` doesn't have named fields for those items?

Reply:
Named fields would be nicer, will fix in a follow-on. Initially I had in mind it was mostly bool/string/w/e with arbitrary name/values but in looking again a lot of it is not.